noir-projects/aztec-nr/aztec/src/oracle/pxe_store.nr

```diff
@@ -0,0 +1,16 @@
+#[oracle(store)]
+unconstrained fn store_oracle<let N: u32>(key: Field, values: [Field; N]) {}
+
+/// Store an array of values in local PXE database. The data is scoped to the current contract.
+pub unconstrained fn store<let N: u32>(key: Field, values: [Field; N]) {
+    store_oracle(key, values);
+}
+
+#[oracle(load)]
+unconstrained fn load_oracle<let N: u32>(key: Field) -> [Field; N] {}
+
+/// Load an array of values from local PXE database. The data is scoped to the current contract.
+/// If the key does not exist, an error is thrown.
+pub unconstrained fn load<let N: u32>(key: Field) -> [Field; N] {
+    load_oracle(key)
+}
```

Comment:
It'd be good to return a boolean from the oracle and wrap the return value in an `Option`: we'll need to be able to handle values not existing in the database.

Reply:
(`load_oracle` would return an array and a boolean, and `load` would return `Option<T> where T: Serialize<N>`)

Reply:
Addressed in newer commits